assets/src/components/Lightbox/Lightbox.js

```diff
@@ -0,0 +1,154 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+import PhotoSwipe from '../../../../node_modules/photoswipe/dist/photoswipe.js';
+import PhotoSwipeUI_Default from '../../../../node_modules/photoswipe/dist/photoswipe-ui-default.js';
+
+export const setupLightboxForImageBlocks = function() {
+  const images = document.querySelectorAll('.wp-block-image img');
+  const items = [...images].map(image => ({
+    src: image.src ? image.src : image.dataset.src,
+    w: 0,
+    h: 0
+  }));
+
+  // We can't use `createPortal` outside `render()`,
+  // `React.render()` needs a node, even if it ends up being empty.
+  // See: https://github.com/facebook/react/issues/12653#issuecomment-382851495
+  const lightBoxNode = document.createElement('div');
+  lightBoxNode.style.position = 'fixed';
+
+  document.body.appendChild(lightBoxNode);
+
+  const imageBlocks = document.querySelectorAll('.wp-block-image');
+  imageBlocks.forEach((imageBlock, index) => {   
+    const caption = imageBlock.querySelector('figcaption');
+    if (caption) {
+      items[index].title = caption.innerHTML;
+    }
+
+    imageBlock.querySelector('img').addEventListener('click', () => wp.element.render(
+        <Lightbox isOpen={true} index={index} items={items} onClose={() => {}} />,
+        lightBoxNode
+      )
+    );
+  })
+
+  return items;
+}
+
+export const wpImagesToItems = images => images.map(
+  image => ({
+    src: image.image_src,
+    w: 0,
+    h: 0,
+  })
+);
+
+export const Lightbox = props => {
+  let pswpElement = useRef(null);
+
+  const options = {
+    index: props.index || 0,
+    closeOnScroll: false,
+    history: false
+  };
+
+  const [currentIndex, setCurrentIndex] = useState(options.index);
+
+  useEffect(() => {
+    const photoSwipe = new PhotoSwipe(pswpElement, PhotoSwipeUI_Default, props.items, options);
+
+    if (photoSwipe) {
```

Comment:
Early return here would make this a lot more readable

Reply:
I missed this one!